### PR TITLE
Add snowflake connector documentation

### PR DESCRIPTION
**Pull request by dulanjalidilmi**
## Purpose
$subject

## Automation tests
 - Unit tests 
   > Code coverage information
 - Integration tests
   > Details about the test cases and coverage

## Security checks
 - Followed secure coding standards in http://wso2.com/technical-reports/wso2-secure-engineering-guidelines? yes/no
 - Ran FindSecurityBugs plugin and verified report? yes/no
 - Confirmed that this PR doesn't commit any keys, passwords, tokens, usernames, or other secrets? yes/no

### CODE REVIEW

File: en/docs/reference/connectors/snowflake-connector/snowflake-connector-example.md [hunk 1]
```diff
@@ -0,0 +1,388 @@
...
+
+* `/insertReservation` : Insert a single reservation to the Snowflake database.
+* `/insertReservationBatch` : Insert a batch of reservations to the Snowflake database.
+* `/getReservationInfo` : Retrieve all the reservations from the Snowflake database.
+* `/deleteReservation` : Delete a reservation from the Snowflake database.
+
+## Before you begin
+
+Before you begin, you must have a valid Snowflake account. 
+To use the Snowflake database, you must have a valid Snowflake account. To create a snowflake account, please visit the official Snowflake website and complete the registration process. Once registered you will obtain username and password which you can login to Snowflake account, and the account identifier which is the unique identifier for your Snowflake account within your business entity and the Snowflake network.
```
Comment (DinithiDiaz) [inline en/docs/reference/connectors/snowflake-connector/snowflake-connector-example.md:24]:
```suggestion
To use the Snowflake database, you must have a valid Snowflake account. To create a snowflake account, please visit the official Snowflake website and complete the registration process. Once registered you will obtain a username and password with which you can log in to your Snowflake account, and the account identifier which is the unique identifier for your Snowflake account within your business entity and the Snowflake network.
```

File: en/docs/reference/connectors/snowflake-connector/snowflake-connector-example.md [hunk 2]
```diff
@@ -0,0 +1,388 @@
...
+
+### Import the connector
+
+Follow these steps to set up the Integration Project and the Connector Exporter Project.
+
+{!includes/reference/connectors/importing-connector-to-integration-studio.md!}
+
+### Add integration logic
+
+First create a REST API called `SnowflakeConnectorApi` in your project
```
Comment (DinithiDiaz) [inline en/docs/reference/connectors/snowflake-connector/snowflake-connector-example.md:57]:
```suggestion
First, create a REST API called `SnowflakeConnectorApi` in your project.
```

File: en/docs/reference/connectors/snowflake-connector/snowflake-connector-example.md [hunk 3]
```diff
@@ -0,0 +1,388 @@
...
+
+| Name | Context |
+| ---------------- | ---------------- |
+| SnowflakeConnectorApi  | /snowflakeconnector  |
+
+Create the following resources in 'SnowflakeConnectorApi' REST API
```
Comment (DinithiDiaz) [inline en/docs/reference/connectors/snowflake-connector/snowflake-connector-example.md:63]:
```suggestion
Create the following resources in the 'SnowflakeConnectorApi' REST API.
```

File: en/docs/reference/connectors/snowflake-connector/snowflake-connector-example.md [hunk 4]
```diff
@@ -0,0 +1,388 @@
...
+
+| uri-template                |
+|-----------------------------| 
+| /insertReservation          |
+| /getReservationInfo         |
+| /insertReservationBatch     |
+| /deleteReservation          |
+
+Let's add the operations to the resources in `SnowflakeConnectorApi` API
```
Comment (DinithiDiaz) [inline en/docs/reference/connectors/snowflake-connector/snowflake-connector-example.md:72]:
```suggestion
Let's add the operations to the resources in the `SnowflakeConnectorApi` API.
```

File: en/docs/reference/connectors/snowflake-connector/snowflake-connector-example.md [hunk 5]
```diff
@@ -0,0 +1,388 @@
...
+
+#### - /insertReservation
+
+Users can utilize this resource to insert a single record into the snowflake database. The user will be sending the reservation payload in the request body.
```
Comment (DinithiDiaz) [inline en/docs/reference/connectors/snowflake-connector/snowflake-connector-example.md:76]:
```suggestion
Users can utilize this resource to insert a single record into the Snowflake database. The user will be sending the reservation payload in the request body.
```

File: en/docs/reference/connectors/snowflake-connector/snowflake-connector-example.md [hunk 6]
```diff
@@ -0,0 +1,388 @@
...
+
+1. In the API insequence drag and drop the [Property Mediator]({{base_path}}/reference/mediators/property-mediator/) to extract the payload from the request body. Let's store the payload in a property named `payload`.
+    ```xml
+      <property expression="json-eval($)" name="payload" scope="default" type="STRING"/>
+   ```
+
+   2. Drag and drop `execute` operation from **SnowflakeConnector** section.
```
Comment (DinithiDiaz) [inline en/docs/reference/connectors/snowflake-connector/snowflake-connector-example.md:83]:
```suggestion
   2. Drag and drop the `execute` operation from the **SnowflakeConnector** section.
```

File: en/docs/reference/connectors/snowflake-connector/snowflake-connector-example.md [hunk 7]
```diff
@@ -0,0 +1,388 @@
...
+      1. Double click the operation. It will show you the properties section.
+      2. In the General Section of the properties, click on the `+` button next to `Snowflake Connection`
```
Comment (DinithiDiaz) [inline en/docs/reference/connectors/snowflake-connector/snowflake-connector-example.md:85]:
```suggestion
      2. In the **General** Section of the properties, click on the `+` button next to `Snowflake Connection`.
```

File: en/docs/reference/connectors/snowflake-connector/snowflake-connector-reference.md [hunk 8]
```diff
@@ -0,0 +1,108 @@
...
+        </tr>
+        <tr>
+            <td>Sql Query</td>
+            <td>SQL query to execute</td>
+            <td>Yes</td>
+        </tr>
+      </table>
+
+??? note "snowflake.execute"
+    The `snowflake.execute` operation execute the given sql statement. This operation can be used to perform DDL and DML Operations.
```
Comment (DinithiDiaz) [inline en/docs/reference/connectors/snowflake-connector/snowflake-connector-reference.md:61]:
```suggestion
    The `snowflake.execute` operation execute the given SQL statement. This operation can be used to perform DDL and DML Operations.
```

File: en/docs/reference/connectors/snowflake-connector/snowflake-connector-reference.md [hunk 9]
```diff
@@ -0,0 +1,108 @@
...
+            <td>Snowflake Account password.</td>
+            <td>Yes</td>
+        </tr>
+    </table>
+
+
+## Operations
+
+??? note "snowflake.query"
+    The `snowflake.query` operation query the given sql statement.
```
Comment (DinithiDiaz) [inline en/docs/reference/connectors/snowflake-connector/snowflake-connector-reference.md:41]:
```suggestion
    The `snowflake.query` operation queries the given SQL statement.
```

File: en/docs/reference/connectors/snowflake-connector/snowflake-connector-example.md (hunk 1 above)
Comment (dulanjalidilmi) [reply en/docs/reference/connectors/snowflake-connector/snowflake-connector-example.md]: Fixed.

File: en/docs/reference/connectors/snowflake-connector/snowflake-connector-example.md (hunk 2 above)
Comment (dulanjalidilmi) [reply en/docs/reference/connectors/snowflake-connector/snowflake-connector-example.md]: Fixed

File: en/docs/reference/connectors/snowflake-connector/snowflake-connector-example.md (hunk 3 above)
Comment (dulanjalidilmi) [reply en/docs/reference/connectors/snowflake-connector/snowflake-connector-example.md]: Fixed.

File: en/docs/reference/connectors/snowflake-connector/snowflake-connector-example.md (hunk 4 above)
Comment (dulanjalidilmi) [reply en/docs/reference/connectors/snowflake-connector/snowflake-connector-example.md]: Fixed.

File: en/docs/reference/connectors/snowflake-connector/snowflake-connector-example.md (hunk 5 above)
Comment (dulanjalidilmi) [reply en/docs/reference/connectors/snowflake-connector/snowflake-connector-example.md]: Fixed.

File: en/docs/reference/connectors/snowflake-connector/snowflake-connector-example.md (hunk 6 above)
Comment (dulanjalidilmi) [reply en/docs/reference/connectors/snowflake-connector/snowflake-connector-example.md]: Fixed.

File: en/docs/reference/connectors/snowflake-connector/snowflake-connector-example.md (hunk 7 above)
Comment (dulanjalidilmi) [reply en/docs/reference/connectors/snowflake-connector/snowflake-connector-example.md]: Fixed.

**Review by DinithiDiaz — CHANGES_REQUESTED** [review]

File: en/docs/reference/connectors/snowflake-connector/snowflake-connector-reference.md (hunk 9 above)
Comment (dulanjalidilmi) [reply en/docs/reference/connectors/snowflake-connector/snowflake-connector-reference.md]: Fixed.

File: en/docs/reference/connectors/snowflake-connector/snowflake-connector-reference.md (hunk 8 above)
Comment (dulanjalidilmi) [reply en/docs/reference/connectors/snowflake-connector/snowflake-connector-reference.md]: Fixed.

File: en/docs/reference/connectors/snowflake-connector/snowflake-connector-overview.md [hunk 10]
```diff
@@ -0,0 +1,27 @@
+# Snowflake Connector Overview
+
+The Snowflake Connector enables you to utilize a complete range of Snowflake operations using WSO2 MI. It provides functionalities to execute a set of standard Snowflake DDL, DML and query commands.
```
[the remark this answers is not in the document]
Comment (dulanjalidilmi) [reply en/docs/reference/connectors/snowflake-connector/snowflake-connector-overview.md]: Fixed.

**Review by DinithiDiaz — APPROVED** [review]

File: en/docs/reference/connectors/snowflake-connector/snowflake-connector-example.md [hunk 11]
```diff
@@ -0,0 +1,388 @@
...
+Follow these steps to deploy the exported CApp in the integration runtime.
+
+{!includes/reference/connectors/deploy-capp.md!}
+
+### Testing the resources
+
+Let's test the API. Deploy the carbon application and start the Micro Integrator.
+
+1. Insert Reservation Resource
+    1. Create a file called payload.json with the following payload.
```
[the remark this answers is not in the document]
Comment (dulanjalidilmi) [reply en/docs/reference/connectors/snowflake-connector/snowflake-connector-example.md]: Fixed.

File: en/docs/reference/connectors/snowflake-connector/snowflake-connector-example.md [hunk 12]
```diff
@@ -0,0 +1,388 @@
...
+## Get the project
+
+You can download the ZIP file and extract the contents to get the project code.
+
+<a href="{{base_path}}/assets/attachments/connectors/snowflake-connector.zip">
+    <img src="{{base_path}}/assets/img/integrate/connectors/download-zip.png" width="200" alt="Download ZIP">
+</a>
+
+!!! tip
+You may need to update the values of Snowflake connection configs before deploying and running this project.
```
[the remark this answers is not in the document]
Comment (dulanjalidilmi) [reply en/docs/reference/connectors/snowflake-connector/snowflake-connector-example.md]: Fixed.

**Pull request merged**